### ISSUE TITLE
[private_cluster] Option to complete terraform at nodepool soak stage during blue-green upgrade

### ISSUE THREAD

**Issue by navilg** [terraform-google-modules/terraform-google-kubernetes-engine#1734]
### TL;DR

A feature to add a variable or configuration using which we can define if terraform should wait for complete upgrade or it can complete its check once worker node upgrade reaches node_pool_soak stage.

If blue green upgrade is enabled, When we change the worker node version and re-run the terraform apply. A worker node upgrade is triggered. With blue-green upgrade enabled, worker node upgrade goes through multiple phase, and each phase varies in time. If batch_soak_period and node pool soak period is large enough, terraform fails with time out waiting for upgrade to complete.

Its also not feasible to keep terraform timeout to the extent of long hours. 

### Terraform Resources

```markdown
https://registry.terraform.io/modules/terraform-google-modules/kubernetes-engine/google/latest/submodules/private-cluster
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster
```

### Additional information

Once cluster upgrade reach to node pool soak stage, that means worker node has already been upgraded. This is the stage GKE waits for users to test and cancel/rollback if something doesnot work as expected. This will not be needed in terraform because this can be cancelled and  rolledback using gcloud command only if something is not working as expected.

**Comment by navilg** [comment]
Hello

**Comment by m0ps** [comment]
@navilg 
I think it's not something that could be controlled by tf module... It's for tf provider (https://github.com/hashicorp/terraform-provider-google/issues) if tf itself allows to implementation of such a trick.

**Comment by navilg** [comment]
Agree. Created an issue there - https://github.com/hashicorp/terraform-provider-google/issues/16770
